installer-clientside/x1p-js/src/x1p.ts

```diff
@@ -81,13 +81,15 @@ export class Printer {
         this.sshPort = 2222;
     }
     
-    mqttRecvAsync<T>(fn: (topic: string, msg: object, resolve: (T) => void, reject: (any) => void) => void) {
+    mqttRecvAsync<T>(filterMapFn: (topic: string, msg: object) => T) {
       const _this = this;
       return new Promise<T>((resolve, reject) => {
         function msg(topic: string, message: Uint8Array) {
-          fn(topic, JSON.parse(new TextDecoder().decode(message)),
-              (x: T) => { _this.mqttClient.off('message', msg); resolve(x); },
-              (x: any) => { _this.mqttClient.off('message', msg); reject(x); });
+          const result = filterMapFn(topic, JSON.parse(new TextDecoder().decode(message)))
+          if (result !== undefined) {
+            _this.mqttClient.off('message', msg);
+            resolve(result);
+          }
```

Comment:
so the reason I originally wrote it this way was so that a filter function could throw if needed, though admittedly I never used this, I suppose

Reply:
The filter function can still throw and it'll reject as its still in that promise body afaik